optaweb-employee-rostering-backend/src/test/java/org/optaweb/employeerostering/roster/RosterRestControllerTest.java

```diff
@@ -446,7 +470,101 @@ public void testPublishAndProvision() {
+        response = getShiftRosterAsExcel(requestSpotList, startDate, endDate);
+
+        assertThat(response.getStatusCode()).as("Response should be successful for two spots").isEqualTo(HttpStatus.OK);
+        try (Workbook workbook = new XSSFWorkbook(new ByteArrayInputStream(response.getBody()))) {
+            assertThat(workbook.getNumberOfSheets()).as("There should only be two sheets in the file").isEqualTo(2);
+            Sheet spotSheet = workbook.getSheet(requestSpotList.get(0).getName());
+            assertSheetForSpotIsCorrect(spotSheet, requestSpotList.get(0), startDate, endDate);
+            spotSheet = workbook.getSheet(requestSpotList.get(1).getName());
+            assertSheetForSpotIsCorrect(spotSheet, requestSpotList.get(1), startDate, endDate);
+        } catch (IOException e) {
+            fail("Unable to read response as an Excel file");
+        }
+
+        Spot badSpot = new Spot(TENANT_ID, "Bad", Collections.emptySet());
+        badSpot.setId(-1L);
+
+        requestSpotList = Arrays.asList(badSpot);
+        response = getShiftRosterAsExcel(requestSpotList, startDate, endDate);
+
+        assertThat(response.getStatusCode()).as("Response should fail with BAD_REQUEST for non-existing spot in tenant")
+                .isEqualTo(HttpStatus.BAD_REQUEST);
+    }
+
+    private void assertSheetForSpotIsCorrect(Sheet spotSheet, Spot spot, LocalDate from, LocalDate to) {
+        assertThat(spotSheet).as("Spot %s sheet should not be null", spot.getName()).isNotNull();
+
+        Map<Long, String> employeeIdToNameList = employeeList.stream()
+                .collect(Collectors.toMap(Employee::getId, Employee::getName));
+
+        Map<List<LocalDateTime>, List<ShiftView>> requestShiftMapByStartAndEndTime = shiftViewList.stream()
+                // Include only shift views for this spot, between from and to
+                .filter(sv -> sv.getSpotId().equals(spot.getId()) &&
+                        // to is exclusive, from is inclusive
+                        !sv.getStartDateTime().toLocalDate().isAfter(to.minusDays(1)) &&
+                        !sv.getEndDateTime().toLocalDate().isBefore(from))
+                .collect(Collectors.groupingBy(sv -> Arrays.asList(sv.getStartDateTime(), sv.getEndDateTime())));
+
+        assertThat(spotSheet.getRow(0).getCell(0).getStringCellValue())
+                .as("Header Column A should be Start").isEqualTo("Start");
+        assertThat(spotSheet.getRow(0).getCell(1).getStringCellValue())
+                .as("Header Column B should be End").isEqualTo("End");
+        assertThat(spotSheet.getRow(0).getCell(2).getStringCellValue())
+                .as("Header Column C should be Employee").isEqualTo("Employee");
+
+        // Sort startEndTimePairs by start time then end time
+        List<List<LocalDateTime>> sortedStartEndTimePairs = requestShiftMapByStartAndEndTime.keySet().stream()
+                .sorted(Comparators.lexicographical(LocalDateTime::compareTo))
```

Comment:
I don't recall adding any dependencies to the `pom.xml`, so any dependency on `Guava` was there before. I might change it depending on how much cleaner the code looks. (code would probably look better with Records (https://openjdk.java.net/jeps/359), but we can't use them until minimum release is Java 14).

Reply:
> I don't recall adding any dependencies to the pom.xml, so any dependency on Guava was there before.

Correct. This is called a transitive dependency (aka undeclared used dependency). You're depending on something that one of your declared dependencies uses but you don't declare it yourself. This is dangerous from the maintenance perspective. Two bad things may happen:
- **You remove** the dependency that transitively provides Guava because you're replacing the technology or dropping the feature. Easy to fix: you declare Guava dependency.
- You or **your parent upgrades** the dependency that transitively provides Guava but the new version no longer depends on it. That will be annoying for the parent maintainer because the FDB will fail surprisingly. They will need to go to this repo and declare the dependency. Wouldn't happen if the dependency was already declared.